map-view/src/common/Map.ts

```diff
@@ -773,23 +775,163 @@ class Map {
+            const options = optionsFromCapabilities(capabilities, {
+              layer: identifier,
+              matrixSet: "ETRS-GK25",
+              // Don't specify style, let it use the default
+            });
+
+            if (!options) {
+              console.error("optionsFromCapabilities returned null. Available layer info:", layerExists);
+              throw new Error(`Could not create WMTS options for layer ${identifier} with ETRS-GK25 matrix set`);
+            }
+
+            console.log(`WMTS options created successfully for layer: ${identifier}`);
+            console.log("Options URL:", options.url);
+            console.log("Options layer:", options.layer);
+            console.log("Options matrixSet:", options.matrixSet);
+            console.log("Options style:", options.style);
+            console.log("Options format:", options.format);
+            console.log("Tile grid origin:", options.tileGrid?.getOrigin(0));
+            console.log("Tile grid resolutions:", options.tileGrid?.getResolutions());
+            console.log("Tile grid extent:", options.tileGrid?.getExtent());
+
+            if (!options.url) {
+              console.log("URL was undefined, setting manually to:", sourceUrl);
+              options.url = sourceUrl;
+            }
+
+            // Remove the restrictive extent from tile grid
+            // The parsed extent is too small and excludes your working tiles
+            // This is kind of a hack to fix it
+            // might be something todo with resolutions and extent mismatch in capabilities
+            // LIIK-857 created to investigate further
+            if (options.tileGrid) {
+              const currentExtent = options.tileGrid.getExtent();
+              console.log("Original restrictive extent:", currentExtent);
+              console.log("Original origin from capabilities:", options.tileGrid.getOrigin(0));
+
+              // Fix the origin using working tile coordinates
+              // Working tile: TileCol=510, TileRow=835 at level 10 (resolution index 10 = 256)
+              // We know this tile should be at world coordinates around Helsinki city center
+              const resolutions = options.tileGrid.getResolutions();
+              const level10Resolution = resolutions[10]; // Should be 256
+              const tileSize = options.tileGrid.getTileSize(0) || 256; // Default to 256 if undefined
+              const worldTileSize = level10Resolution * (typeof tileSize === "number" ? tileSize : 256);
+
+              console.log("Level 10 resolution:", level10Resolution);
+              console.log("Tile size:", tileSize);
+              console.log("World tile size:", worldTileSize);
+
+              // Calculate origin from working tile coordinates
+              // If tile [510, 835] should be at Helsinki center coordinates [25499052, 6675851]
+              const helsinkiCenterX = 25499052;
```

Comment:
While magic numbers can be excused when documented, I do not understand what was the process for getting these `helsinkiCenterX`, `helsinkiCenterY`, `offsetCorrectionX`, `offsetCorrectionY`, `510` or `835`.

Can you explain where these numbers come from? Is it from some specific query to some endpoint? Did you click around and take notes? Were they in some discussion document?